pymongo/common.py

```diff
@@ -817,9 +821,13 @@ def validate_auth_option(option: str, value: Any) -> tuple[str, Any]:
 def validate(option: str, value: Any) -> tuple[str, Any]:
     """Generic validation function."""
     lower = option.lower()
-    validator = VALIDATORS.get(lower, raise_config_error)
-    value = validator(option, value)
-    return option, value
+    try:
+        validator = VALIDATORS[lower]
+        value = validator(option, value)
+        return option, value
+    except KeyError:
+        suggestions = get_close_matches(lower, VALIDATORS, cutoff=0.2)
+        raise_config_error(option, suggestions)
```

Comment:
What do you think about refactoring this with the code block below? It seems like they are essentially identical besides the VALIDATORS vs URI_OPTIONS_VALIDATOR_MAP.

Reply:
I've added a note /hint on camelCase and suggested referring to docstring.
I've also opened a ticket to explore rationalizing the keys to kwargs: https://jira.mongodb.org/browse/PYTHON-4056

Reply:
Still todo.

Reply:
Hi. If the original intent hasn't been resolved, would you please elaborate?

Reply:
Can you refactor this to remove the duplicated code so that there's only one call to try/except + get_close_matches/raise_config_error, eg:
```python
def _get_validator(name, validators):
    lower = name.lower()
    try:
        return validators[lower]
    except KeyError:
        suggestions = get_close_matches(lower, validators, cutoff=0.2)
        raise_config_error(name, suggestions)
```